README.md

```diff
@@ -34,136 +38,128 @@ yarn add react-onesignal
 
 ```js
 import OneSignal from 'react-onesignal';
 
-OneSignal.initialize('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', options);
-```
-
-Where options is:
-
-```ts
-safari_web_id?: string;
-subdomainName?: string;
-allowLocalhostAsSecureOrigin?: boolean;
-requiresUserPrivacyConsent?: boolean;
-persistNotification?: boolean;
-autoResubscribe?: boolean;
-autoRegister?: boolean;
-notificationClickHandlerMatch?: string;
-notificationClickHandlerAction?: string;
-notifyButton?: {
-  enable?: boolean;
-  size?: 'small' | 'medium' | 'large';
-  position?: 'bottom-left' | 'bottom-right';
-  showCredit?: boolean;
-  prenotify?: boolean;
-    theme?: 'default' | 'inverse';
-    offset?: {
-      bottom?: string;
-      right?: string;
-      left?: string;
-    },
-    text?: {
-      [key: string]: string;
-    };
-    colors?: {
-      [key: string]: string;
-    };
-}
-```
-
-And `OneSignal` object contains:
-
-```ts
-notificationPermission: string[];
-registerForPushNotifications: () => Promise<any>;
-getNotificationPermission: () => Promise<string>;
-isPushNotificationsEnabled: () => Promise<boolean>;
-isPushNotificationsSupported: () => boolean;
-setSubscription: (unmute: boolean) => Promise<any>;
-setEmail: (email: string) => Promise<string>;
-logoutEmail: () => Promise<void>;
-getEmailId: () => Promise<string>;
-getUserId: () => Promise<string>;
-setExternalUserId: (externalUserId: string | number) => Promise<void>;
-removeExternalUserId: () => Promise<void>;
-getExternalUserId: () => Promise<any>;
-initialized: boolean;
-sendTag: (key: string, val: string) => Promise<string>;
-sendTags: (keyValues: object) => Promise<any>;
+OneSignal.initialize('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx');
 ```
 
-## Advanced Usage
-
-### Player ID
-
-Player ID is an important information on OneSignal.
-
-You can use `getPlayerId` to obtain it.
+The `initialize` function returns a promise that resolves when OneSignal is loaded.
 
+**Examples**
 ```js
-// Obtains the current playerId from the browser
-const playerId = await OneSignal.getPlayerId();
+await OneSignal.initialize('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx');
+// do other stuff
 ```
-
-### Notification Permission Management
-
-To manage notifications:
-
-- Use `notificationPermission` to see all possible states.
-- Use `getNotificationPermission` to check current notification permissions.
-- Use `registerForPushNotifications` to ask for notification permissions.
+---
 
 ```js
-// Check all possible permission states
-const permissions = OneSignal.notificationPermission;
-
-// Check current permission state
-const currentState = await OneSignal.getNotificationPermission();
-
-// Ask the user for notification permissions, if not granted yet
-await OneSignal.registerForPushNotifications();
+OneSignal.initialize('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx').then(() => {
+  this.setState({ initialized: true });
+  OneSignal.showSlidedownPrompt().then(() => {
+    // do other stuff
+  });
+})
```

Comment:
Maybe we should write this in functional component paradigm, in order to keep up with the latest react versions?

E.g.

```suggestion
const [initialized, setInitialized] = useState(false);

OneSignal.initialize('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx').then(() => {
  setInitialized(true);
  OneSignal.showSlidedownPrompt().then(() => {
    // do other stuff
  });
})
```

Reply:
Marking as resolved since this was commited